### PR TITLE
Update to vsjdbc 12.0.0: changes to character sets / Fix CVEs

### LINKED ISSUE #42: Update vs-common-jdbc to v12.0.0 for Exasol V8 changes + adapt tests

### LINKED ISSUE #43: 🔐 CVE-2024-25710: org.apache.commons:commons-compress:jar:1.24.0:test
## Summary
Loop with Unreachable Exit Condition ('Infinite Loop') vulnerability in Apache Commons Compress.This issue affects Apache Commons Compress: from 1.3 through 1.25.0.

Users are recommended to upgrade to version 1.26.0 which fixes the issue.

CVE: CVE-2024-25710
CWE: CWE-835

## References
- https://ossindex.sonatype.org/vulnerability/CVE-2024-25710?component-type=maven&component-name=org.apache.commons%2Fcommons-compress&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
- http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2024-25710
- https://lists.apache.org/thread/cz8qkcwphy4cx8gltn932ln51cbtq6kf

### LINKED ISSUE #44: 🔐 CVE-2024-26308: org.apache.commons:commons-compress:jar:1.24.0:test
## Summary
Allocation of Resources Without Limits or Throttling vulnerability in Apache Commons Compress.This issue affects Apache Commons Compress: from 1.21 before 1.26.

Users are recommended to upgrade to version 1.26, which fixes the issue.

CVE: CVE-2024-26308
CWE: CWE-770

## References
- https://ossindex.sonatype.org/vulnerability/CVE-2024-26308?component-type=maven&component-name=org.apache.commons%2Fcommons-compress&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
- http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2024-26308
- https://lists.apache.org/thread/ch5yo2d21p7vlqrhll9b17otbyq4npfg
- https://www.openwall.com/lists/oss-security/2024/02/19/2

### PR DESCRIPTION
**Pull request by pj-spoelders**
The behaviour when it comes to character sets is now simplified,
The target char set is now always UTF-8.
The `IMPORT_DATA_TYPES` property (and value `FROM_RESULT_SET`) are now deprecated (change in vs-common-jdbc):
An exception will be thrown when users use`FROM_RESULT_SET`. The exception message warns the user that the value is no longer supported and the property itself is also deprecated.

We also updated dependencies and resolved the following 2 CVEs:
- CVE-2024-26308: org.apache.commons:commons-compress:jar:1.24.0:test
- CVE-2024-25710: org.apache.commons:commons-compress:jar:1.24.0:test

Closes #42 
Closes #43 
Closes #44

### CODE REVIEW

File: src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java [hunk 1]
```diff
@@ -400,7 +400,8 @@ void testSelectAllColsNumberFromOra() throws SQLException {
         void testSelectAllColsNumberFromOraWithJDBCTypemapping() throws SQLException {
             final String qualifiedTableNameActual = VIRTUAL_SCHEMA_ORACLE_NUMBER_TO_DECIMAL_JDBC_MAPPING + "."
                     + TABLE_ORACLE_NUMBER_HANDLING;
-            // previously final ResultSet expected = getExpectedResultSet("(A VARCHAR(100), B VARCHAR(100), C
+            // previously final ResultSet expected = getExpectedResultSet("(A VARCHAR(100), B VARCHAR(100),
+            // C
             // VARCHAR(100))",
```
Comment (kaklakariada) [inline src/test/java/com/exasol/adapter/dialects/oracle/OracleSqlDialectIT.java:405]:
```suggestion
```

File: pom.xml [hunk 2]
```diff
@@ -33,6 +33,13 @@
             <artifactId>error-reporting-java</artifactId>
             <version>1.0.1</version>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/com.exasol/maven-project-version-getter -->
+        <dependency>
+            <groupId>com.exasol</groupId>
+            <artifactId>maven-project-version-getter</artifactId>
+            <version>1.2.0</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (kaklakariada) [inline pom.xml:42]: Is this used in the code? I didn't see changes using it.

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: package-lock.json
[the remark this answers is not in the document]
Comment (pj-spoelders) [reply package-lock.json:1]: Let me check

File: pom.xml (hunk 2 above)
Comment (pj-spoelders) [reply pom.xml:42]: It's used in the ExtensionIT class

File: package-lock.json
[the remark this answers is not in the document]
Comment (pj-spoelders) [reply package-lock.json:1]: Seems that's not the case .. maybe a finger slip. Removed and resolved

File: .gitattributes [hunk 3]
```diff
@@ -2,10 +2,11 @@ pk_generated_parent.pom                                          linguist-genera
...
+.github/workflows/ci-build.yml                                   linguist-generated=true
 .github/workflows/ci-build-next-java.yml                         linguist-generated=true
 .github/workflows/dependencies_check.yml                         linguist-generated=true
 .github/workflows/release_droid_print_quick_checksum.yml         linguist-generated=true
-
+.github/workflows/release_droid_upload_github_release_assets.yml linguist-generated=true
+.github/workflows/release_droid_prepare_original_checksum.yml    linguist-generated=true
 .settings/org.eclipse.jdt.core.prefs linguist-generated=true
 .settings/org.eclipse.jdt.ui.prefs   linguist-generated=true
-extension/package-lock.json          linguist-generated=true
```
[the remark this answers is not in the document]
Comment (pj-spoelders) [reply .gitattributes:11]: Added it again

File: .project-keeper.yml [hunk 4]
```diff
@@ -11,7 +11,7 @@ version:
   fromSource: pom.xml
 build:
   runnerOs: ubuntu-20.04
-  freeDiskSpace: false
+  freeDiskSpace: true
   exasolDbVersions:
     - "7.1.25"
     - "8.24.0"
```
Comment (kaklakariada) [inline .project-keeper.yml:17]:
```suggestion
    - "8.25.0"
```

File: doc/changes/changes_3.0.0.md [hunk 5]
```diff
@@ -1,12 +1,26 @@
-# Oracle Virtual Schema 3.0.0, released 2024-??-??
+# Oracle Virtual Schema 3.0.0, released 2024-02-22
 
-Code name:
+Code name: Char set is always `utf-8`, deprecated IMPORT_DATA_TYPES `FROM_RESULT_SET` value .
```
Comment (kaklakariada) [inline doc/changes/changes_3.0.0.md:3]:
```suggestion
Code name: Char set is always `utf-8`, deprecated IMPORT_DATA_TYPES `FROM_RESULT_SET` value
```

File: doc/changes/changes_3.0.0.md [hunk 6]
```diff
@@ -1,12 +1,26 @@
...
 
 ## Summary
 
+The behaviour when it comes to character sets is now simplified,
+The target char set is now always UTF-8.
+The `IMPORT_DATA_TYPES` property (and value `FROM_RESULT_SET`) are now deprecated (change in vs-common-jdbc):
+An exception will be thrown when users use`FROM_RESULT_SET`. The exception message warns the user that the value is no longer supported and the property itself is also deprecated.
```
Comment (kaklakariada) [inline doc/changes/changes_3.0.0.md:10]:
```suggestion
An exception will be thrown when users use `FROM_RESULT_SET`. The exception message warns the user that the value is no longer supported and the property itself is also deprecated.
```

File: doc/changes/changes_3.0.0.md [hunk 7]
```diff
@@ -1,12 +1,26 @@
...
+
+We also updated dependencies and resolved the following 2 CVEs:
+- CVE-2024-26308: org.apache.commons:commons-compress:jar:1.24.0:test
+- CVE-2024-25710: org.apache.commons:commons-compress:jar:1.24.0:test
+
 ## Features
 
-* ISSUE_NUMBER: description
+* #42: Update vs-common-jdbc to v12.0.0 for Exasol V8 changes + adapt tests refactoring
```
Comment (kaklakariada) [inline doc/changes/changes_3.0.0.md:18]:
```suggestion
* #42: Updated vs-common-jdbc to v12.0.0 for Exasol V8 changes + adapt tests refactoring
```

File: doc/changes/changes_3.0.0.md [hunk 8]
```diff
@@ -1,12 +1,26 @@
...
+
+## Security
+
+* #43: CVE-2024-25710: org.apache.commons:commons-compress:jar:1.24.0:test
+* #44: CVE-2024-26308: org.apache.commons:commons-compress:jar:1.24.0:test
```
Comment (kaklakariada) [inline doc/changes/changes_3.0.0.md:23]:
```suggestion
* #43: Fixed CVE-2024-25710 in `org.apache.commons:commons-compress:jar:1.24.0:test`
* #44: Fixed CVE-2024-26308 in `org.apache.commons:commons-compress:jar:1.24.0:test`
```

File: doc/changes/changes_3.0.0.md [hunk 9]
```diff
@@ -1,12 +1,26 @@
...
 ## Summary
 
+The behaviour when it comes to character sets is now simplified,
+The target char set is now always UTF-8.
+The `IMPORT_DATA_TYPES` property (and value `FROM_RESULT_SET`) are now deprecated (change in vs-common-jdbc):
+An exception will be thrown when users use`FROM_RESULT_SET`. The exception message warns the user that the value is no longer supported and the property itself is also deprecated.
+
+We also updated dependencies and resolved the following 2 CVEs:
+- CVE-2024-26308: org.apache.commons:commons-compress:jar:1.24.0:test
+- CVE-2024-25710: org.apache.commons:commons-compress:jar:1.24.0:test
```
Comment (kaklakariada) [inline doc/changes/changes_3.0.0.md:14]:
```suggestion
We also updated dependencies and resolved the following 2 CVEs in test dependency `org.apache.commons:commons-compress`:
- CVE-2024-26308
- CVE-2024-25710
```

File: src/test/java/com/exasol/adapter/dialects/oracle/OracleScalarFunctionsIT.java [hunk 10]
```diff
@@ -153,4 +147,14 @@ public Connection createExasolConnection() throws SQLException {
...
+
+    @Override
+    protected void beforeAllSetup() throws SQLException {
+        TimeZone.setDefault(TimeZone.getTimeZone("UTC"));
+    }
+
+    @Override
+    protected void afterAllTeardown() throws SQLException {
+
+    }
```
Comment (kaklakariada) [inline src/test/java/com/exasol/adapter/dialects/oracle/OracleScalarFunctionsIT.java:159]:
```suggestion
```
Is it possible to remove this when it's empty? If this is required please add a short comment with an explanation

**Review by kaklakariada — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/adapter/dialects/oracle/OracleScalarFunctionsIT.java (hunk 10 above)
Comment (pj-spoelders) [reply src/test/java/com/exasol/adapter/dialects/oracle/OracleScalarFunctionsIT.java:159]: No, it is an abstract protected method

Comment (pj-spoelders) [reply src/test/java/com/exasol/adapter/dialects/oracle/OracleScalarFunctionsIT.java]: I added comments

**Review by kaklakariada — DISMISSED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**